src/Microsoft.PowerApps.TestEngine/TestEngine.cs

```diff
@@ -88,10 +88,26 @@ public async Task<string> RunTestAsync(FileInfo testConfigFile, string environme
                     throw new ArgumentNullException(nameof(tenantId));
                 }
 
+                if (string.IsNullOrEmpty(domain))
+                {
+                    throw new ArgumentNullException(nameof(domain));
+                }
+
                 if (outputDirectory == null)
                 {
                     throw new ArgumentNullException(nameof(outputDirectory));
                 }
+                else
+                {
+                    var inputUri = new Uri(outputDirectory.FullName.StartsWith(@"\\?\") ? outputDirectory.FullName.Replace(@"\\?\", "") : outputDirectory.FullName);
+                    if (!new Uri(_fileSystem.GetDefaultRootTestEngine()).IsBaseOf(inputUri))
+                    {
+                        var wrongLocationError = $"Please ensure {nameof(outputDirectory)} is set to a value resolving to a location inside the permitted output location.";
```

Comment:
Put {nameof(outputDirectory)} in quotes:

\'{nameof(outputDirectory)}'\

Reply:
added